pymeasure/instruments/lecroy/lecroyT3DSO1204.py

```diff
@@ -0,0 +1,1324 @@
+                  from the trigger waveform. Use AC coupling to get a stable edge trigger when
+                  your waveform has a large dc offset.
+        • dc    — DC coupling allows dc and ac signals into the trigger path.
+        • lowpass  — HFREJ coupling places a lowpass filter in the trigger path.
+        • highpass — LFREJ coupling places a highpass filter in the trigger path.
+        """,
+        validator=strict_discrete_set,
+        values={"ac": "AC", "dc": "DC", "lowpass": "HFREJ", "highpass": "LFREJ"},
+        map_values=True
+    )
+
+    trigger_level = Instrument.control(
+        "TRLV?", "TRLV %.2EV",
+        """ A float parameter that sets the trigger level voltage for the active trigger source.
+            When there are two trigger levels to set, this command is used to set the higher
+            trigger level voltage for the specified source. :attr:`trigger_level2` is used to set
+            the lower trigger level voltage.
+            When setting the trigger level it must be divided by the probe attenuation. This is
+            not documented in the datasheet and it is probably a bug of the scope firmware.
+            An out-of-range value will be adjusted to the closest legal value.
+        """
+    )
+
+    trigger_level2 = Instrument.control(
+        "TRLV2?", "TRLV2 %.2EV",
+        """ A float parameter that sets the lower trigger level voltage for the specified source.
+        Higher and lower trigger levels are used with runt/slope triggers.
+        When setting the trigger level it must be divided by the probe attenuation. This is
+        not documented in the datasheet and it is probably a bug of the scope firmware.
+        An out-of-range value will be adjusted to the closest legal value.
+        """
+    )
+
+    trigger_slope = Instrument.control(
+        "TRSL?", "TRSL %s",
+        """ A string parameter that sets the trigger slope of the specified trigger source.
+        <trig_slope>:={NEG,POS,WINDOW} for edge trigger.
+        <trig_slope>:={NEG,POS} for other trigger
+        """,
+        validator=strict_discrete_set,
+        values={"negative": "NEG", "positive": "POS", "window": "WINDOW"},
+        map_values=True
+    )
+
+    def values(self, command, **kwargs):
+        """ Reads a set of values from the instrument through the adapter,
+        passing on any key-word arguments.
+        """
+        return self.instrument.values("C%d:%s" % (self.number, command), **kwargs)
+
+    def ask(self, command):
+        return self.instrument.ask("C%d:%s" % (self.number, command))
+
+    def write(self, command):
+        # only in case of the BWL command the syntax is different. Why? SIGLENT Why?
+        if command == "BWL OFF":
+            self.instrument.write("BWL C%d,OFF" % self.number)
+        elif command == "BWL ON":
+            self.instrument.write("BWL C%d,ON" % self.number)
+        else:
+            self.instrument.write("C%d:%s" % (self.number, command))
```

Comment:
If you subclass instrument.Channel, you can use `insert_id` to change the command syntax for write, ask and values:

```suggestion
    def insert_id(self, command):
        # only in case of the BWL command the syntax is different. Why? SIGLENT Why?
        if command == "BWL OFF":
            return "BWL C%d,OFF" % self.number
        elif command == "BWL ON":
            return "BWL C%d,ON" % self.number
        else:
            return "C%d:%s" % (self.number, command)
```

Reply:
Same as the previous comment.

Reply:
Very neat. Thank you.